File: subs/insert_signals.py
"""Insert signals for 이효석 videos"""
import httpx

BASE = "https://arypzhotxflimroprmdk.supabase.co/rest/v1/"
KEY = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6ImFyeXB6aG90eGZsaW1yb3BybWRrIiwicm9sZSI6ImFub24iLCJpYXQiOjE3NzIwMDYxMTAsImV4cCI6MjA4NzU4MjExMH0.qcqFIvYRiixwu609Wjj9H3HxscU8vNpo9nS_KQ3f00A"
HEADERS = {
    "apikey": KEY,
    "Authorization": f"Bearer {KEY}",
    "Content-Type": "application/json",
    "Prefer": "return=representation"
}

VIDEO1_ID = "cfc9e60f-5bae-45c5-8e57-8f624f79298c"
VIDEO2_ID = "94ff3a67-01d6-49da-89ff-d461a810774c"
SPEAKER_ID = "b07d8758-493a-4a51-9bc5-7ef75f0be67c"

signals = [
    # Video 1: fDZnPoK5lyc
    {
        "video_id": VIDEO1_ID,
        "speaker_id": SPEAKER_ID,
        "stock": "HD현대일렉트릭",
        "market": "KR",
        "signal": "긍정",
        "mention_type": "결론",
        "confidence": "medium",
        "timestamp": "08:12",
        "key_quote": "AI 인프라의 핵심 파트너가 돼 버렸다. 아주 강력한 모트",
        "reasoning": "전세계 5개 회사만 가능한 초고압 변압기 기술. 영업이익률 계속 상승. 다만 명확한 매수 권유 아님, 전망 형태.",
        "pipeline_version": "V9.1",
        "review_status": "pending",
    },
    {
        "video_id": VIDEO1_ID,
        "speaker_id": SPEAKER_ID,
        "stock": "효성중공업",
        "market": "KR",
        "signal": "긍정",
        "mention_type": "결론",
        "confidence": "medium",
        "timestamp": "08:15",
        "key_quote": "HD현대일렉트릭이나 효성중공업이 AI 인프라의 핵심 파트너가 돼 버렸다",
        "reasoning": "초고압 변압기 제조 가능 기업. 극도로 높은 진입장벽. 전망 형태이나 매수 권유 아님.",
        "pipeline_version": "V9.1",
        "review_status": "pending",
    },
    {
        "video_id": VIDEO1_ID,
        "speaker_id": SPEAKER_ID,
        "stock": "조선",
        "market": "SECTOR",
        "signal": "긍정",
        "mention_type": "결론",
        "confidence": "medium",
        "timestamp": "08:53",
        "key_quote": "아직 식지 않은 기대감. 단순 하청이 아닌 안보 파트너로 격상",
        "reasoning": "한국 조선사 실적 좋고 고수익 선박으로 전환. 미국 브릿지 전략 발표로 안보 파트너 격상. LNG 운반선 수요 증가.",
        "pipeline_version": "V9.1",
        "review_status": "pending",
    },
    {
        "video_id": VIDEO1_ID,
        "speaker_id": SPEAKER_ID,
        "stock": "현대차",
        "market": "KR",
        "signal": "긍정",
        "mention_type": "결론",
        "confidence": "medium",
        "timestamp": "15:26",
        "key_quote": "자동차 회사가 로봇 회사로 바뀌면서 멀티플이 바뀌는 어마어마한 그런게 있다",
        "reasoning": "자동차→로봇 전환으로 멀티플 변화 기대. 보스턴다이나믹스 R&D→대량생산 전환. GTC 행사 모멘텀.",
        "pipeline_version": "V9.1",
        "review_status": "pending",
    },
    # Video 2: tSXkj2Omz34
    {
        "video_id": VIDEO2_ID,
        "speaker_id": SPEAKER_ID,
        "stock": "삼성전자",
        "market": "KR",
        "signal": "긍정",
        "mention_type": "결론",
        "confidence": "high",
        "timestamp": "05:01",
        "key_quote": "영업이익 43조에서 201조. 이익이 다섯배 늘어난 회사는 본 적이 없어요",
        "reasoning": "영업이익 5배 증가 전망. 글로벌 시총 매달권 진입 가능성. 멀티플 9배 미만으로 극도로 저평가. 다만 '숫자를 따져보자' 형태로 전망.",
        "pipeline_version": "V9.1",
        "review_status": "pending",
    },
    {
        "video_id": VIDEO2_ID,
        "speaker_id": SPEAKER_ID,
        "stock": "SK하이닉스",
        "market": "KR",
        "signal": "긍정",
        "mention_type": "결론",
        "confidence": "high",
        "timestamp": "05:01",
        "key_quote": "하이닉스 멀티플 다섯배. 이게 말이 돼? 글로벌 3위 진입 가능성",
        "reasoning": "영업이익 47.4조 전망. 멀티플 5배로 글로벌 대비 극도로 저평가. 엔비디아보다 이익 더 나는데 시총 0.4조달러.",
        "pipeline_version": "V9.1",
        "review_status": "pending",
    },
    {
        "video_id": VIDEO2_ID,
        "speaker_id": SPEAKER_ID,
        "stock": "반도체",
        "market": "SECTOR",
        "signal": "긍정",
        "mention_type": "결론",
        "confidence": "high",
        "timestamp": "01:55",
        "key_quote": "반도체가 바뀌었는데 시장에서 제대로 평가를 못 받고 있다",
        "reasoning": "코스피 영업이익 증가분 중 반도체가 100% 기여(210조). 이익 비중 56% 역대 최고. 멀티플 7배→12배 정상화 시 코스피 7900 가능.",
        "pipeline_version": "V9.1",
        "review_status": "pending",
    },
]

ok = 0
for s in signals:
    r = httpx.post(BASE + "influencer_signals", headers=HEADERS, json=s)
    if r.status_code in (200, 201):
        print(f"OK: {s['stock']} ({s['signal']})")
        ok += 1
    else:
        print(f"FAIL: {s['stock']} - {r.status_code} {r.text[:200]}")

print(f"\n=== {ok}/{len(signals)} signals inserted ===")

File: insert_signals.mjs
import fs from 'fs';

const BASE = 'https://arypzhotxflimroprmdk.supabase.co/rest/v1';
const K = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6ImFyeXB6aG90eGZsaW1yb3BybWRrIiwicm9sZSI6InNlcnZpY2Vfcm9sZSIsImlhdCI6MTc3MjAwNjExMCwiZXhwIjoyMDg3NTgyMTEwfQ.Q4ycJvyDqh-3ns3yk6JE4hB2gKAC39tgHE9ofSn0li8';
const H = { 'apikey': K, 'Authorization': `Bearer ${K}`, 'Content-Type': 'application/json' };

const CHANNEL_ID = 'd153b75b-1843-4a99-b49f-c31081a8f566';
const SPEAKER_ID = 'b07d8758-493a-4a51-9bc5-7ef75f0be67c';

async function get(table, params) {
  const r = await fetch(`${BASE}/${table}${params}`, { headers: H });
  return r.json();
}

async function post(table, body, headers = {}) {
  const r = await fetch(`${BASE}/${table}`, {
    method: 'POST', headers: { ...H, 'Prefer': 'return=minimal', ...headers }, body: JSON.stringify(body)
  });
  if (!r.ok) { const t = await r.text(); throw new Error(`${r.status}: ${t}`); }
  return r;
}

// Step 1: Get all existing videos for this channel
console.log('Fetching existing videos...');
const existingVids = await get('influencer_videos', `?channel_id=eq.${CHANNEL_ID}&select=id,video_id&limit=1000`);
const vidMap = new Map(existingVids.map(v => [v.video_id, v.id]));
console.log(`Existing videos: ${vidMap.size}`);

// Step 2: Parse JSONL
const lines = fs.readFileSync('C:\\Users\\Mario\\work\\hs_analysis_cleaned.jsonl', 'utf8').trim().split('\n');
const parsed = lines.map(l => JSON.parse(l));

// Step 3: Create missing videos
const missingVids = parsed.filter(p => !vidMap.has(p.vid_id));
console.log(`Missing videos to create: ${missingVids.length}`);

for (let i = 0; i < missingVids.length; i += 50) {
  const batch = missingVids.slice(i, i + 50).map(p => ({
    channel_id: CHANNEL_ID,
    video_id: p.vid_id,
    title: p.title || null,
    has_subtitle: false
  }));
  try {
    await post('influencer_videos', batch, { 'Prefer': 'return=representation' });
  } catch (e) {
    // Try one by one for conflicts
    for (const v of batch) {
      try { await post('influencer_videos', v); } catch (e2) { /* skip existing */ }
    }
  }
}

// Re-fetch all videos
const allVids = await get('influencer_videos', `?channel_id=eq.${CHANNEL_ID}&select=id,video_id&limit=1000`);
const vm = new Map(allVids.map(v => [v.video_id, v.id]));
console.log(`Total videos now: ${vm.size}`);

// Step 4: Build signals
const allSignals = [];
for (const obj of parsed) {
  if (!obj.signals) continue;
  const videoUuid = vm.get(obj.vid_id);
  if (!videoUuid) { console.error(`No UUID for ${obj.vid_id}`); continue; }
  for (const s of obj.signals) {
    const ticker = s.ticker || '';
    const conf = s.confidence || 5;
    allSignals.push({
      video_id: videoUuid,
      speaker_id: SPEAKER_ID,
      stock: s.stock || null,
      ticker: ticker || null,
      market: /^\d{6}$/.test(ticker) ? 'KR' : (ticker ? 'US' : null),
      signal: s.signal_type || null,
      confidence: conf <= 3 ? 'low' : conf <= 5 ? 'medium' : conf <= 7 ? 'high' : 'very_high',
      key_quote: s.key_quote || null,
      reasoning: s.reasoning || null,
      mention_type: '논거',
      timestamp: s.timestamp || null,
      review_status: 'pending'
    });
  }
}
console.log(`Total signals to insert: ${allSignals.length}`);

// Step 5: Batch insert
const BATCH = 50;
let success = 0, failed = 0;
const failures = [];

for (let i = 0; i < allSignals.length; i += BATCH) {
  const batch = allSignals.slice(i, i + BATCH);
  let ok = false;
  for (let retry = 0; retry < 3; retry++) {
    try {
      await post('influencer_signals', batch);
      ok = true;
      break;
    } catch (e) {
      console.error(`Batch ${i} retry ${retry}: ${e.message.slice(0, 200)}`);
      if (retry < 2) await new Promise(r => setTimeout(r, 2000));
    }
  }
  if (ok) success += batch.length;
  else { failed += batch.length; failures.push({ from: i, count: batch.length }); }
  
  const done = success + failed;
  if (done % 100 < BATCH || i + BATCH >= allSignals.length) {
    console.log(`Progress: ${done}/${allSignals.length} (${(done/allSignals.length*100).toFixed(1)}%)`);
  }
}

console.log(`\n=== COMPLETE ===`);
console.log(`channel_id: ${CHANNEL_ID}`);
console.log(`speaker_id: ${SPEAKER_ID}`);
console.log(`Success: ${success} / Failed: ${failed}`);
if (failures.length) console.log('Failures:', JSON.stringify(failures));
